internal/repository/master_index.go

```diff
@@ -114,31 +146,37 @@ func (mi *MasterIndex) Remove(index *Index) {
 }
 
 // Store remembers the id and pack in the index.
-func (mi *MasterIndex) Store(pb restic.PackedBlob) {
+func (mi *MasterIndex) StorePack(id restic.ID, blobs []restic.Blob) {
 	mi.idxMutex.Lock()
 	defer mi.idxMutex.Unlock()
 
+	// delete blobs from pending
+	for _, blob := range blobs {
+		mi.pendingBlobs.Delete(restic.BlobHandle{Type: blob.Type, ID: blob.ID})
+	}
+
 	for _, idx := range mi.idx {
 		if !idx.Final() {
-			idx.Store(pb)
+			idx.StorePack(id, blobs)
 			return
 		}
 	}
 
 	newIdx := NewIndex()
-	newIdx.Store(pb)
+	newIdx.StorePack(id, blobs)
 	mi.idx = append(mi.idx, newIdx)
 }
 
-// NotFinalIndexes returns all indexes that have not yet been saved.
-func (mi *MasterIndex) NotFinalIndexes() []*Index {
+// NotFinalIndexes finalizes all indexes that have not yet been saved and returns that list
```

Comment:
The method name changed, but the comment doesn't reflect that.

Reply:
is changed now.